datafusion/optimizer/src/single_distinct_to_groupby.rs

```diff
@@ -257,6 +257,56 @@ impl OptimizerRule for SingleDistinctToGroupBy {
                                 )))
                             }
                         }
+                        Expr::AggregateFunction(AggregateFunction {
```

Comment:
From what I can tell this is now a general optimization (as in it will rewrite *any* distinct user defined aggregate  as well). If so, that is quite cool. Is it correct?

Reply:
Can we please add a test (if not already done) showing that the distinct aggregate is rewritten? Or perhaps there is already a test for median 🤔